apps/tlv-app/grails-app/assets/javascripts/menus/annotations-map.js

```diff
@@ -397,146 +421,318 @@ function saveAnnotations() {
-					$( "#dragoImage" ).attr( "src", imageUrl );
-				});
+					var imageUrl   = urlCreator.createObjectURL( blob );
+					$( '#dragoImage' ).attr( 'src', imageUrl );
+				} );
 			}
 		);
 		tlv.map.renderSync();
-
-		displayLoadingDialog( "Saving..." );
+		
+		displayLoadingDialog( 'Saving...' );
 	}
 }
 
-function searchForAnnotations() {
-	$.each( tlv.layers, function( index, layer ) {
-		if ( !layer.annotations ) {
-			var params = {
-				filter: "image_id LIKE '" + layer.imageId + "'",
-				maxResults: 100,
-				outputFormat: "JSON",
-				request: "getFeature",
-				service: "WFS",
-				typeName: "omar:annotation",
-				version: "1.1.0"
-			};
-			$.ajax({
-				url: tlv.libraries[ layer.library ].wfsUrl + "?" + $.param( params )
-			})
-			.done(function( data ) {
-				layer.annotations = data;
-				if ( data.features.length > 0 ) {
-					addAnnotations( data, layer );
-				}
-				searchForAnnotations();
-			})
-			.fail( function() {
-				searchForAnnotations();
-			});
+// function searchForAnnotations() {
+// 	console.log( 'searchForAnnotations' );
+//
+// 	$.each( tlv.layers, function( index, layer ) {
+// 			console.log( layer );
+//
+// 			var params = {
+// 				filter: "image_id LIKE '" + layer.imageId + "'",
+// 				maxResults: 100,
+// 				outputFormat: "JSON",
+// 				request: "getFeature",
+// 				service: "WFS",
+// 				typeName: "omar:annotation",
+// 				version: "1.1.0"
+// 			};
+//
+// 			$.ajax( {
+// 				url: tlv.libraries[ layer.library ].wfsUrl + "?" + $.param( params )
+// 			} )
+// 			.done( function( data ) {
+// 				console.log( data );
+//
+// 				layer.annotations = data;
+// 				if ( data.features.length > 0 ) {
+// 					addAnnotations( data, layer );
+// 				}
+// 				// searchForAnnotations();
+// 			})
+// 			.fail( function() {
+// 				// searchForAnnotations();
+// 			});
+//
+//
+// 			return false;
+// 	});
+// }
+
+function addSavedAnnotations( geoJsonSource, layer ) {
+	var features = new ol.format.GeoJSON().readFeatures( geoJsonSource );
+	
+	$.each( features, function( index, feature ) {
+		var geometry = new ol.format.WKT().readGeometry(
+			feature.getProperties().geometry_ortho, {
+				dataProjection: 'EPSG:4326',
+				featureProjection: 'EPSG:3857'
+			} );
+		
+		feature.setGeometry( geometry );
+	} );
+	
+	createAnnotationsLayer( layer );
+	
+	var source = layer.annotationsLayer.getSource();
+	
+	source.un( 'addfeature', anAnnotationHasBeenAdded );
+	
+	layer.annotationsLayer.getSource().addFeatures( features );
+	
+	source.on( 'addfeature', anAnnotationHasBeenAdded );
+	
+	$.each( source.getFeatures(), function( index, feature ) {
+		var properties = feature.getProperties();
+		var style      = properties.style || createDefaultStyle();
+		var label      = properties.type + '\n' + properties.username;
+		
+		style.getText().setText( label );
+		feature.setStyle( style );
+	} );
+}
 
+function generateRandomNumber( min, max ) {
+	return ~~( Math.random() * max ) + min;
```

Comment:
```suggestion
	return Math.floor(Math.random() * (max - min) ) + min;
```

Reply:
Remove floor() if you don't want an integer. The max - min math avoids the possibility of generating a number greater than the MAX if min=max